src/index.js

```diff
@@ -15,7 +16,9 @@ ReactDOM.render(
     <Router>
       <AuthProvider>
         <NoteProvider>
-          <App />
+          <FilterProvider>
+            <App />
+          </FilterProvider>
         </NoteProvider>
       </AuthProvider>
```

Comment:
And instead of adding the providers in index.js you can create a new file called as CombinedProviders.jsx

```
import React from "react";
import { AuthProvider, NoteProvider,FilterProvider } from "LOCATION WHERE CONTEXTS ARE STORED";

const CombinedProvider = ({ children }) => {
    return (
        <AuthProvider>
        <NoteProvider>
          <FilterProvider>
            {children}
          </FilterProvider>
        </NoteProvider>
      </AuthProvider>
    )
}

export { CombinedProvider };

```
And then you can start using it in your index.js file just shifting your work from one file to another

Reply:
That's a welcome suggestion. Will implement this during Refactor.